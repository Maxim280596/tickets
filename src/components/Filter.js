import React from 'react';
import {Button, Col, Row} from 'react-bootstrap';
import styled from 'styled-components';


const StyledButton = styled.button `
  width:50%;
  height:25%;
  margin-bottom: 20px;
  border-radius:10px;
  
`;


export  function Filter () {


  return(
    < >
    
    <StyledButton variant="primary" size="lg">Самый дешевый</StyledButton>
    <StyledButton variant="primary" size="lg">Самый быстрый</StyledButton>
    
    </>
  )
}