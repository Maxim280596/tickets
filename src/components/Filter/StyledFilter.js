import styled from "styled-components";

export const StyledButtonPrice = styled.button`
  width: 50%;
  min-height: 50px;
  margin-bottom: 20px;
  border-radius: 10px 0 0 10px;
  background-color: white;
  border: 1px solid #e0e0e0;
  box-shadow: 1px 1px 4px 1px #e0e0e0;
  &:hover {
    background-color: #2196f3;
    color:white;
  }
`;

export const StyledButtonFast = styled.button`
  width: 50%;
  min-height: 50px;
  margin-bottom: 20px;
  border-radius: 0 10px 10px 0;
  background-color: white;
  box-shadow: 1px 1px 4px 1px #e0e0e0;
  border: 1px solid #e0e0e0;
  &:hover {
    background-color: #2196f3;
    color:white;
  }
`;
