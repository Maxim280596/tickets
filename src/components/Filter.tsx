import React, { useState } from 'react';
import { useDispatch } from 'react-redux';

import { StyledButtonFast, StyledButtonPrice } from '../styled/StyledFilter';
import {
  filteringFast,
  filteringPrice,
} from '../redux/ducks/ticketsReduser';


const Filter = () => {
  const dispatch = useDispatch();
  const [price, setPrice] = useState(false);
  const [fast, setFast] = useState(false);
  const filterPrice = () => {
    dispatch(filteringPrice());
    setPrice(true);
    setFast(false);
  };

  const filterFast = () => {
    dispatch(filteringFast());
    setPrice(false);
    setFast(true);
  };

  return (
    <>
      <StyledButtonPrice onClick={filterPrice} active={price}>
        САМЫЙ ДЕШЕВЫЙ
      </StyledButtonPrice>
      <StyledButtonFast onClick={filterFast} active={fast}>
        САМЫЙ БЫСТРЫЙ
      </StyledButtonFast>
    </>
  );
};

export default Filter;
