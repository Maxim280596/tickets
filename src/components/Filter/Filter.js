import React from "react";
import { StyledButtonFast, StyledButtonPrice } from "./StyledFilter";

export function Filter() {
  return (
    <div>
      <StyledButtonPrice variant="primary" size="lg">
        САМЫЙ ДЕШЕВЫЙ
      </StyledButtonPrice>
      <StyledButtonFast variant="primary" size="lg">
        САМЫЙ БЫСТРЫЙ
      </StyledButtonFast>
    </div>
  );
}
