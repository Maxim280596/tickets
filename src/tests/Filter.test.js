import React from 'react';
import {render} from 'enzyme';
import toJson from 'enzyme-to-json';

import Filter from '../components/Filter';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

describe('Testing <Filter/>', () => {
  const initialState = {output:10}
  const mockStore = configureStore()
  let store
  it('Filter have rendered correctly', ()=>{
    store = mockStore(initialState)
    const filter = render(<Provider store={store}><Filter/></Provider>);
    expect(toJson(filter)).toMatchSnapshot();  
  })
})