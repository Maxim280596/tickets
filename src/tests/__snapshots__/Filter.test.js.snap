// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing rendering of <Filter/> should Filter have be rendered correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Filter
    onClick={[MockFunction]}
  >
    <styled.button
      active={false}
      className="price"
      onClick={[Function]}
    >
      <button
        className="sc-bdfBwQ eJidin price"
        onClick={[Function]}
      >
        САМЫЙ ДЕШЕВЫЙ
      </button>
    </styled.button>
    <styled.button
      active={false}
      className="fast"
      onClick={[Function]}
    >
      <button
        className="sc-gsTCUz kzxJJP fast"
        onClick={[Function]}
      >
        САМЫЙ БЫСТРЫЙ
      </button>
    </styled.button>
  </Filter>
</Provider>
`;
